### PR TITLE
(launch) fix: replace hardcode with form value

### PR DESCRIPTION
**Pull request by Tanya-atatakai**
## Fixes
Fix hardcoded validation 

https://github.com/user-attachments/assets/6d028bbc-3de2-4912-9e94-902a0250e286

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: apps/launch/components/MyStakingContracts/Create/useFieldRules.tsx [hunk 1]
```diff
@@ -21,96 +21,102 @@ const getGenericFieldRules = (label: string) => [
...
+        rules: [
+          ...getGenericFieldRules(FieldConfig.rewardsPerSecond.name),
+          {
+            // BE validation from https://github.com/valory-xyz/autonolas-registries/blob/main/contracts/staking/StakingVerifier
+            validator: async (_, value) => {
+              if (!minStakingDepositLimit) return Promise.resolve();
+              if (!apyLimit) return Promise.resolve();
 
-            const rewardsPerYear = value * ONE_YEAR;
-            const apy = (rewardsPerYear * 1e18) / 1000;
```
Comment (Tanya-atatakai) [inline apps/launch/components/MyStakingContracts/Create/useFieldRules.tsx:54]: the `1000` was hardcoded preventing new contracts from being created. We should use the current `minStakingDeposit` value from the form instead

File: apps/launch/components/MyStakingContracts/Create/useFieldRules.tsx [hunk 2]
```diff
@@ -21,96 +21,102 @@ const getGenericFieldRules = (label: string) => [
...
+              const minStakingDeposit = form.getFieldValue('minStakingDeposit');
```
Comment (Tanya-atatakai) [inline apps/launch/components/MyStakingContracts/Create/useFieldRules.tsx:68]: @kupermind you can verify only next three lines

File: apps/launch/components/MyStakingContracts/Create/index.spec.tsx [hunk 3]
```diff
@@ -120,6 +120,12 @@ describe('<CreateStakingContract />', () => {
     await userEvent.type(rewardsPerSecondInput, '100');
     expect(rewardsPerSecondInput).toHaveValue('100');
 
+    const minStakingDepositInput = screen.getByLabelText('Minimum service staking deposit, OLAS');
+    // clear first, because it has default value
+    await userEvent.clear(minStakingDepositInput);
+    await userEvent.type(minStakingDepositInput, '1000');
+    expect(minStakingDepositInput).toHaveValue('1000');
```
Comment (kupermind) [inline apps/launch/components/MyStakingContracts/Create/index.spec.tsx:127]: Why is this needed at all? Can it be just read from the contract every time?

Comment (Tanya-atatakai) [reply apps/launch/components/MyStakingContracts/Create/index.spec.tsx:127]: it's just a test......

**Review by kupermind — APPROVED** [review]

**Review by truemiller — APPROVED** [review]
